aequilibrae/paths/traffic_assignment.py

```diff
@@ -512,6 +514,68 @@ def results(self) -> pd.DataFrame:
 
         return df
 
+    def save_turning_volumes(
+        self,
+        table_name: str,
+        turns_df: pd.DataFrame,
+        classes: Optional[list[str]] = None,
+        iteration: Optional[int] = None,
+        blend_iterations: bool = True,
+    ) -> pd.DataFrame:
+        """Saves the turning movements to results_database.sqlite
+
+        Method fails if table exists
+
+        Args:
+            table_name (:obj:`str`): Name of the table to hold this assignment result
+            turns_df (:obj:`pd.DataFrame`): Dataframe containing a, b, c nodes of required turning volumes.
+            classes: (:obj:`list`): List containing requires traffic classes.
+                If None, all classes used in assignment will be exported.
+            iteration: (:obj:`int`): Desired iteration.
+            blend_iterations: (:obj:`bool`): whether or not to blend iterations.
+        """
+        df = self.turning_volumes(turns_df, classes, iteration, blend_iterations)
+        conn = sqlite3.connect(path.join(self.project.project_base_path, "results_database.sqlite"))
+        df.to_sql(table_name, conn)
+        conn.close()
+
+        conn = self.project.connect()
+        data = [table_name, "turn volumes", self.procedure_id, "", self.procedure_date, self.description]
+        conn.execute(
+            """Insert into results(table_name, procedure, procedure_id, procedure_report, timestamp,
+                                            description) Values(?,?,?,?,?,?)""",
+            data,
+        )
+        conn.commit()
+        conn.close()
+        return df
+
+    def turning_volumes(
+        self,
+        turns_df: pd.DataFrame,
+        classes: Optional[list[str]] = None,
```

Comment:
This type hint indicates that a `str` is ok, but the `TurnVolumesResults.from_traffic_class` expects a traffic class object
```diff
-         classes: Optional[list[str]] = None,
+         classes: Optional[list[TrafficClass]] = None,
```